xarray_beam/_src/core_test.py

```diff
@@ -278,5 +278,46 @@ def test_dataset_to_chunks_vars(self):
     self.assertIdenticalChunks(actual, expected)
 
 
+class ValidateChunksTest(test_util.TestCase):
+
+  def test_unmatched_dimension_raises_error(self):
+    dataset = xarray.Dataset({'foo': ('x', np.arange(6))})
+    with self.assertRaises(ValueError) as e:
+      (
+        test_util.EagerPipeline()
+        | beam.Create([(xbeam.Key({'x': 0, 'y': 0}), dataset)])
```

Comment:
Passing in a list is a nice alternative to EagerPipeline/Create:
```suggestion
        [(xbeam.Key({'x': 0, 'y': 0}), dataset)]
```